task_processing/plugins/mesos/execution_framework.py

```diff
@@ -253,18 +259,18 @@ def get_tasks_to_launch(self, offer):
         with self._lock:
             # Get all the tasks of the queue
             while not self.task_queue.empty():
-                task = self.task_queue.get()
+                task, mesos_task_id = self.task_queue.get()
 
                 if ((remaining_cpus >= task.cpus and
                      remaining_mem >= task.mem and
                      remaining_disk >= task.disk and
                      remaining_gpus >= task.gpus)):
                     # This offer is sufficient for us to launch task
                     tasks_to_launch.append(
-                        self.create_new_docker_task(
-                            offer,
-                            task,
-                            available_ports
+                        (
+                            self.create_new_docker_task(
+                                offer, task, mesos_task_id, available_ports),
+                            mesos_task_id
```

Comment:
mesos_task_id is the task_id field of TaskInfo. I don't see a need for (task_id, taskInfo) tuple instead of just TaskInfo.